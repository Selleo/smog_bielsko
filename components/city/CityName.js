import React, { Component, PropTypes } from 'react';
import { StyleSheet, Text, View } from 'react-native';

export default class CityName extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <View>
        <Text style={styles.text}>
          {this.props.text}
        </Text>
      </View>
    )
  }

  static propTypes = {
    text: PropTypes.string.isRequired,
  };
}

const styles = StyleSheet.create({
  text: {
    fontSize: 30,
    color: '#363636'
  }
});
