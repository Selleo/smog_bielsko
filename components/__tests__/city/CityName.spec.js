import 'react-native';
import React                          from 'react';

import { shallow }                    from 'enzyme';
import chai, { expect }               from 'chai';
import dirtyChai                      from 'dirty-chai';

import CityName                       from '../../city/CityName';

const originalExpect = global.expect;

chai.use(dirtyChai);

let wrapper, props;

describe('CityName', () => {
  it('renders correctly', () => {
    props = { text: 'Zywiec' };
    wrapper = shallow(<CityName {...props}/>);

    originalExpect(wrapper).toMatchSnapshot();
    expect(wrapper).to.exist();
  });

  it('renders description on html', () => {
    props = { text: 'Zywiec' };
    wrapper = shallow(<CityName {...props}/>);
    expect(wrapper.children().length).to.equal(1);

    expect(wrapper.children().children().text()).to.equal('Zywiec');
  });

});
