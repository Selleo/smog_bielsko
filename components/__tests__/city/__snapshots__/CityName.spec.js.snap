exports[`CityName renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View>
    <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
            Object {
                "color": "#363636",
                "fontSize": 30,
              }
        }>
        Zywiec
    </Text>
</View>,
  "nodes": Array [
    <View>
      <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                  Object {
                        "color": "#363636",
                        "fontSize": 30,
                      }
            }>
            Zywiec
      </Text>
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <CityName
        text="Zywiec" />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": CityName {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "text": "Zywiec",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                              Object {
                                        "color": "#363636",
                                        "fontSize": 30,
                                      }
                    }>
                    Zywiec
          </Text>
</View>,
        "_debugID": 2,
        "_renderedOutput": <View>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                              Object {
                                        "color": "#363636",
                                        "fontSize": 30,
                                      }
                    }>
                    Zywiec
          </Text>
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CityName
    text="Zywiec" />,
}
`;
